game-store-service/functions/catalogBatchProcess.ts

```diff
@@ -0,0 +1,110 @@
+import { SNS } from "aws-sdk";
+import {closeClient, invokeClient} from "./invokeClient";
+
+const { SNS_ARN } = process.env;
+
+type ProductCreationStatus = "SUCCESS" | "FAILURE";
+
+const catalogBatchProcess = async event => {
+  let client;
+  try {
+    client = await invokeClient();
+    const products = event.Records.map(({ body }) => JSON.parse(body));
+
+    await Promise.all(
+      products.map(async (product) => {
+        let status: ProductCreationStatus = "SUCCESS";
+
+        try {
+          await addProductToDB(client, product);
+        } catch (error) {
+          status = "FAILURE";
+          console.error("DB error: ", error?.message);
+        }
+
+        try {
+          await publishMessageToSNS(product, status);
+        } catch (error) {
+          console.error("SNS error: \n", JSON.stringify(error));
+        }
+      })
+    );
+  } catch (error) {
+    console.error("Error during catalogBatchProcess: \n", JSON.stringify(error));
+  } finally {
+    closeClient(client);
+  }
+};
+
+const publishMessageToSNS = ({ title }, status: ProductCreationStatus) => {
+  const sns = new SNS();
+
+  return sns
+    .publish({
+      Subject: "Product creation result",
+      Message:
+        status === "SUCCESS"
+          ? `The following product has been added to product DB:\n${title}`
+          : `The following product has not been added to product DB:\n${title}`,
+      TopicArn: SNS_ARN,
+      MessageAttributes: {
+        status: {
+          DataType: "String",
+          StringValue: status,
+        },
+      },
+    })
+    .promise();
+};
+
+const addProductToDB = async (client, product) => {
+  const { title, studio, genre, description, release_date, poster, price, count } = formatProduct(product);
+
+  if (!isProductValid({ title, price, count })) {
+    throw new Error(`Product ${title} is not valid`);
+  }
+
+  const {
+    rows: [{ id }],
+  } = await client.query(
+    'insert into products(title, studio, genre, description, release_date, poster, price) values ($1, $2, $3, $4, $5, $6, $7) RETURNING *',
+    [title, studio, genre, description, release_date, poster, price]
+  );
+  await client.query(
+    'insert into stocks(product_id, count) values ($1, $2)',
+    [id, count]
+  );
```

Comment:
Consider using transactions there.
Also it would be better inserting all products in one query instead of using separate queries for each item.

Reply:
I'll rework this function before merge.
But as a part of task it doesn't affect on score.

Anyway, thanks for advice.